### PR TITLE
feature: configurable routes

### PR DESCRIPTION
**Pull request by anondev2323**
- [x] Add routes as a configurable option
       - Check if a route is in the list before performing actions
- [x] Add `getMinAmount` onto routes
- [x] Clean up Route.Bridge/Route.Relay checks across application
       - Left over from when there were only the 2 routes.  Now we check each route dynamically
- [x] Add logic for determining the default route (if bridge is not available)
- [x] Filter networks by available routes
- [x] Only query available routes for available tokens, do not query for a token if it's already in the supported list
- [x] Add validation for route

Bonus:
- [x] Add `.tsx` files to `tsconfig.json` and fix type errors
- [x] Clean up config folder, how we define the constants in `config/index.ts` and validation checks
       - Rename "network" to "chain". Network refers to mainnet, testnet and devnet and chain refers to the individual blockchains
       - Split up chains/tokens/gasEstimates into their own files so they're easier to locate
- [x] Clean up the gas estimation logic, it had gotten kind of out of hand
       - Gas fallback can be accessed using the chainName and route
       - Gas estimates were moved to the chain contexts so that we got rid of all of the `if/else`s
       - If gas cannot be estimated, throw an error.  Catch the error and grab the fallback estimate
- [x] Optimize `allSupportedSourceTokens` and `allSupportedDestTokens`.  Don't need to query every token on every route.  Only need to query if we know it's not supported by another route.
- [x] Fix gas formatting bug which was causing EVM gas estimates to appear as 0

### CODE REVIEW

File: wormhole-connect/src/utils/routes/cctpManual.ts [hunk 1]
```diff
@@ -623,7 +634,8 @@ export class CCTPManualRoute extends BaseRoute {
   }: TransferDestInfoParams): Promise<TransferDisplayData> {
     const token = TOKENS[txData.tokenKey];
     const { gasToken } = CHAINS[txData.toChain]!;
-    const gas = await calculateGas(txData.toChain, Route.CCTPManual, receiveTx);
+    // const gas = await calculateGas(txData.toChain, Route.CCTPManual, receiveTx);
+    const gas = 'TODO';
```
Comment (anondev2323) [inline wormhole-connect/src/utils/routes/cctpManual.ts]: Missed this

File: wormhole-connect/src/utils/routes/cosmosGateway.ts [hunk 2]
```diff
@@ -734,11 +741,12 @@ export class CosmosGatewayRoute extends BaseRoute {
...
-      txData.toChain,
-      Route.COSMOS_GATEWAY,
-      receiveTx,
-    );
+    // const gas = await calculateGas(
+    //   txData.toChain,
+    //   Route.CosmosGateway,
+    //   receiveTx,
+    // );
+    const gas = 'TODO';
```
Comment (anondev2323) [inline wormhole-connect/src/utils/routes/cosmosGateway.ts]: Missed this

File: wormhole-connect/src/utils/routes/operator.ts [hunk 3]
```diff
@@ -141,13 +137,90 @@ export default class Operator {
...
+  async allSupportedDestTokens(
+    sourceToken: TokenConfig | undefined,
+    sourceChain?: ChainName | ChainId,
+    destChain?: ChainName | ChainId,
+  ): Promise<TokenConfig[]> {
+    const supported: { [key: string]: TokenConfig } = {};
+    for (const route of ROUTES) {
+      for (const key in TOKENS) {
+        const alreadySupported = supported[key];
+        if (!alreadySupported) {
```
Comment (anondev2323) [inline wormhole-connect/src/utils/routes/operator.ts:191]: We were querying all tokens for every route.  Tried to cut that down a bit by checking if they were already on the list

File: wormhole-connect/src/views/Bridge/Inputs/AmountInput.tsx [hunk 4]
```diff
@@ -23,8 +23,14 @@ function AmountInput(props: Props) {
...
-  function handleAmountChange(event) {
-    let value = event.target.value;
+  function handleAmountChange(
+    e:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>
+      | undefined,
+  ) {
+    let value = e!.target.value;
+    // let value = event.target.value;
```
Comment (anondev2323) [inline wormhole-connect/src/views/Bridge/Inputs/AmountInput.tsx]: Need to delete random comment

**Comment by derpy-duck** [comment]
' Add logic for determining the default route (if bridge is not available)

Where is this change done?

**Comment by anondev2323** [comment]
> ' Add logic for determining the default route (if bridge is not available)
> 
> Where is this change done?

So it just sets the default to the first route (`ROUTES[0]`).  And then it relies on `establishRoute` in Bridge.tsx.  I also added a config validation to ensure that at least 1 route is enabled.  It was too much of a headache to set the route to a falsey value

**Comment by anondev2323** [comment]
Just realizing, the gas estimate when sending is 0, due to a bug with formatting the amount.  Looks like this was present before, not anything introduced by the refactor.  Will push a fix in a moment

**Pull request closed without merging**